network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractIdentifiableImpl.java

```diff
@@ -118,8 +122,8 @@ public void addAlias(String alias, String aliasType, boolean ensureAliasUnicity)
             resource.getAttributes().getAliasesWithoutType().add(uniqueAlias);
         }
         getNetwork().getIdByAlias().put(uniqueAlias, this.getId());
-        getNetwork().getResource().getAttributes().updateResource();
-        resource.getAttributes().updateResource();
+        getNetwork().updateResource();
+        updateResource();
```

Comment:
Maybe a method addAlias(alias, id) and removeAlias(alias) in the network
and call updateResource in these methods ?

Reply:
Done